### PR TITLE
Smaller header files

### PR DESCRIPTION
**Pull request by pirapira**
I'm trying to make compilation units smaller.  Currently I'm removing `#include`s from header files.  Then I had to add some `#include`s in cpp files, but that should be fine.

I also moved one function definition from `State.h` to `State.cpp`.  In this case (a template instantiated in only two different ways), I guess it makes sense to do so.

### CODE REVIEW

File: libdevcore/Common.cpp [hunk 1]
```diff
@@ -31,7 +31,9 @@ namespace dev
 
 char const* Version = ETH_PROJECT_VERSION;
 
+const bytes NullBytes;
```
Comment (gumb0) [inline libdevcore/Common.cpp]: While we're here let's change it to `bytes const` according to coding standards.
Nest lines, too.

File: libdevcore/Common.h [hunk 2]
```diff
@@ -69,7 +69,7 @@ namespace dev
 
 extern char const* Version;
 
-static const std::string EmptyString;
+extern const std::string EmptyString;
```
Comment (gumb0) [inline libdevcore/Common.h]: `std::string const`

File: libdevcore/CommonIO.h [hunk 3]
```diff
@@ -82,7 +81,7 @@ template <class T> inline std::ostream& operator<<(std::ostream& _out, std::vect
 template <class T, std::size_t Z> inline std::ostream& operator<<(std::ostream& _out, std::array<T, Z> const& _e);
 template <class T, class U> inline std::ostream& operator<<(std::ostream& _out, std::pair<T, U> const& _e);
 template <class T> inline std::ostream& operator<<(std::ostream& _out, std::list<T> const& _e);
-template <class T1, class T2, class T3> inline std::ostream& operator<<(std::ostream& _out, std::tuple<T1, T2, T3> const& _e);
+// template <class T1, class T2, class T3> inline std::ostream& operator<<(std::ostream& _out, std::tuple<T1, T2, T3> const& _e);
```
Comment (gumb0) [inline libdevcore/CommonIO.h]: Let's remove this?

File: libdevcore/Log.h [hunk 4]
```diff
@@ -167,17 +156,17 @@ class LogOutputStreamBase
 	void append(u160 const& _t) { m_sstr << EthNavy << _t << EthReset; }
 	void append(double _t) { m_sstr << EthBlue << _t << EthReset; }
 	template <unsigned N> void append(FixedHash<N> const& _t) { m_sstr << EthTeal "#" << _t.abridged() << EthReset; }
-	void append(h160 const& _t) { m_sstr << EthRed "@" << _t.abridged() << EthReset; }
-	void append(h256 const& _t) { m_sstr << EthCyan "#" << _t.abridged() << EthReset; }
-	void append(h512 const& _t) { m_sstr << EthTeal "##" << _t.abridged() << EthReset; }
+	void append(h160 const& _t) { m_sstr << EthRed "@" << _t.h160::abridged() << EthReset; }
```
Comment (gumb0) [inline libdevcore/Log.h]: Why is this change needed?

File: libdevcore/Log.h [hunk 5]
```diff
@@ -268,6 +261,8 @@ class LogOutputStream: LogOutputStreamBase
 
 	LogOutputStream& operator<<(LogTag _t) { m_logTag = _t; return *this; }
 
+	LogOutputStream& operator<<(const char* _t) { if (Id::verbosity <= g_logVerbosity) { if (_AutoSpacing && m_sstr.str().size() && m_sstr.str().back() != ' ') m_sstr << " "; append(_t); } return *this; }
```
Comment (gumb0) [inline libdevcore/Log.h]: Do we need this? Wouldn't `append` overload for `const char*` above be enough for this?

File: libdevcore/RangeMask.h [hunk 6]
```diff
@@ -94,8 +94,8 @@ class RangeMask
 	/// reference to itself.
 	RangeMask& invert() { return *this = inverted(); }
 
-	template <class S> RangeMask operator-(S const& _m) const { auto ret = *this; return ret -= _m; }
-	template <class S> RangeMask& operator-=(S const& _m) { return invert().unionWith(_m).invert(); }
+	RangeMask operator-(RangeMask const& _m) const { auto ret = *this; return ret -= _m; }
```
Comment (gumb0) [inline libdevcore/RangeMask.h]: Looks like we won't be able to do `-` and `-=` with a single number after this... Maybe not a problem, but one convenient usage less

File: libdevcore/Common.cpp (hunk 1 above)
Comment (pirapira) [inline libdevcore/Common.cpp]: By nesting, you mean adding indentation?

File: libdevcore/Log.h (hunk 4 above)
Comment (pirapira) [inline libdevcore/Log.h]: That was a partial attempt to resolve templates manually, but I don't do this in this PR.

File: libdevcore/Log.h (hunk 5 above)
Comment (pirapira) [inline libdevcore/Log.h]: This one, too, was resolving templates manually.  This one is worse so I revert this.

File: libdevcore/RangeMask.h (hunk 6 above)
Comment (pirapira) [inline libdevcore/RangeMask.h]: Suddenly I discovered this `RangeMask` is never used.  I'll remove it all.

File: libdevcore/Common.cpp (hunk 1 above)
Comment (gumb0) [inline libdevcore/Common.cpp]: I meant "next lines", sorry

File: libdevcore/RangeMask.h (hunk 6 above)
Comment (gumb0) [inline libdevcore/RangeMask.h]: Please don't remove it yet, I use it in fast-sync branch

File: libdevcore/RangeMask.h [hunk 7]
```diff
@@ -230,8 +230,7 @@ template <class T> inline std::ostream& operator<<(std::ostream& _out, RangeMask
 	return _out;
 }
 
-template <class T>
-RangeMask<T>& RangeMask<T>::unionWith(typename RangeMask<T>::Range const& _m)
+RangeMask& RangeMask::unionWith(typename RangeMask::Range const& _m)
```
Comment (gumb0) [inline libdevcore/RangeMask.h]: I think `typename` may be removed here now, since RangeMask is not a template

File: libdevcore/TrieHash.h [hunk 8]
```diff
@@ -21,8 +21,9 @@
 
 #pragma once
 
-#include "Common.h"
-#include "FixedHash.h"
+#include <vector>
```
Comment (gumb0) [inline libdevcore/TrieHash.h]: `#include`s for stdlib should be after ones for our libraries, let's do it right it at least for new code

File: libdevcore/Log.h [hunk 9]
```diff
@@ -240,6 +229,10 @@ class LogOutputStreamBase
 		append(_t.second);
 		m_sstr << EthPurple ")" EthReset;
 	}
+	template <class T> void append(const char* _t)
```
Comment (gumb0) [inline libdevcore/Log.h]: Not sure if we still need it in this PR, but it should be either non-template or specialization of `append` template below

File: libdevcore/RangeMask.h [hunk 10]
```diff
@@ -1,294 +0,0 @@
-/*
```
Comment (gumb0) [inline libdevcore/RangeMask.h]: Retuurn it 😭

**Comment by pirapira** [comment]
The AppVeyor build timeouts.  Maybe I should remove more until it passes.

**Comment by gumb0** [comment]
It's probably the tests that take the majority of AppVeyor time.

(I've heard that we have a setting of 2 hours time for AppVeyor but it sometimes gets reset to 1 hour and then it timeouts. But I don't have access to this setting)
